### PR TITLE
Replace deprecated VSTS task

### PR DESCRIPTION
**Pull request by chcosta**
VSTS has deprecated the "Copy and Publish Build Artifacts" task - https://docs.microsoft.com/en-us/vsts/build-release/tasks/utility/copy-and-publish-build-artifacts

We recently hit some instability with the task, and as a first step to improving the reliability, I'm updating to the supported VSTS tasks.

It's difficult to test this change outside of official builds, and that made me a bit nervous, so I flipped the bit on these which says "don't block on failure" for the copy and publish steps.  If they prove to be reliable, we can remove that change, but I don't think it's worth failing a successful release build because of log publishing.

The diff is kind of ugly because VSTS is constantly changing their API.

https://github.com/dotnet/core-eng/issues/1856

### CODE REVIEW

**Comment by chcosta** [comment]
FYI, these failures can't be related to my changes.

File: buildpipeline/DotNet-CoreFx-Trusted-Linux-Crossbuild.json [hunk 1]
```diff
@@ -1,11 +1,13 @@
 {
   "build": [
     {
+      "environment": {},
       "enabled": true,
       "continueOnError": false,
       "alwaysRun": false,
       "displayName": "Change permissions to agent folder for cleanup steps",
       "timeoutInMinutes": 0,
+      "refName": "Task1",
```
Comment (joperezr) [inline buildpipeline/DotNet-CoreFx-Trusted-Linux-Crossbuild.json]: small question: where these task names added automatically or is there a reason why we need them now?

File: buildpipeline/DotNet-CoreFx-Trusted-Linux-Crossbuild.json [hunk 2]
```diff
@@ -179,30 +199,57 @@
...
+        "Parallel": "false",
+        "ParallelCount": "8"
       }
     },
     {
+      "environment": {},
       "enabled": true,
       "continueOnError": true,
-      "alwaysRun": true,
+      "alwaysRun": false,
```
Comment (joperezr) [inline buildpipeline/DotNet-CoreFx-Trusted-Linux-Crossbuild.json]: Why this change? Don't we need to cleanup docker any more?

File: buildpipeline/DotNet-CoreFx-Trusted-Linux-Crossbuild.json [hunk 3]
```diff
@@ -179,30 +199,57 @@
...
+        "CleanTargetFolder": "false",
+        "OverWrite": "false",
+        "flattenFolders": "false"
+      }
+    },
+    {
+      "environment": {},
+      "enabled": true,
+      "continueOnError": true,
+      "alwaysRun": false,
```
Comment (joperezr) [inline buildpipeline/DotNet-CoreFx-Trusted-Linux-Crossbuild.json]: Doesn't this have to always run?

**Comment by joperezr** [comment]
In general it's hard to review these king of changes, but stuff LGTM. I suppose you'll fix the official build in case this breaks the build for any reason?

File: buildpipeline/DotNet-CoreFx-Trusted-Linux-Crossbuild.json (hunk 1 above)
Comment (chcosta) [reply buildpipeline/DotNet-CoreFx-Trusted-Linux-Crossbuild.json]: added automatically

File: buildpipeline/DotNet-CoreFx-Trusted-Linux-Crossbuild.json (hunk 2 above)
Comment (chcosta) [reply buildpipeline/DotNet-CoreFx-Trusted-Linux-Crossbuild.json]: I'm not certain where this change came from.
